### ISSUE TITLE
Query Loop block: Reuse existing `wp-a11y` package `speak()` instead of custom implementaiton.

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#55312]
### Description

The Query Loop block 'enhanced pagination' settings uses its own custom implementation of an aria-live region to deliver audible messages to screen reader users.

If possible, it should reuse the existing `wp-a11y` package script. I do see some blocks just enqueue required core scripts on the front end, for example the comments block. 

The custom implementation in the block:
- Is redundant code, both in the block php file and in the JavaScript view file.
- Comes with reduced functionality, ad the custom implementation doesn't implement all the accessibility features of `wp-a11y`.
- Introduces an unnecessary maintenance cost.
- While the `wp-a11y` packages is supposed to be a stable, tested, implementation, the custom implementation may not be sufficiently tested.

I'm not aware of what the best practice is when a block `view` JavaScript files needs some core package as a dependency. I'm not sure such a mechanism is in place but it appears that in the future there will be more and more the need for the view files to enqueue some dependency scripts.

For now, I only noticed that some blocks do use `wp_enqueue_script()`. For example, the 'Comments' and 'Comments Form' blocks just do this:

`wp_enqueue_script( 'comment-reply' );`

I'd think the Query Loop block should just do the same by enqueueing hte `wp-a11y` script and just call `window.wp.a11y.speak()` where necessary.

### Step-by-step reproduction instructions

See the unnecessary custom implementation of an aria-live region in the block code.

PHP
https://github.com/WordPress/gutenberg/blob/727d3da84398ad6bd47f9a82c42acb1fea4ad55d/packages/block-library/src/query/index.php#L27-L54

JS
https://github.com/WordPress/gutenberg/blob/727d3da84398ad6bd47f9a82c42acb1fea4ad55d/packages/block-library/src/query/view.js#L43-L62

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Marking as bug because the custom implementation is not feature-pair with the wp-a11y package.
- It does not use an `aria-relevant` attribute.
- It does not use an `aria-atomic` attribute.
- It doesn't prepend the visually hidden text `Notifications` before the aria live region.
- **It does not strip HTML tags that could be incorrectly added to the translatable strings.**

**Comment by Mamaduka** [comment]
We might want to reduce the size of the `a11y` package before using it on the front end - https://bundlephobia.com/package/@wordpress/a11y@3.43.0.

Most of the size is due to the `i18n` dependency, which is only used for [a single string](https://github.com/WordPress/gutenberg/blob/27d7cf855c6b73dc863a4613c318c59862378fd6/packages/a11y/src/add-intro-text.js#L19).

**Comment by afercia** [comment]
Good point @Mamaduka thanks.

**Comment by ntsekouras** [comment]
Thanks for the report! The custom live region was removed from the block in https://github.com/WordPress/gutenberg/pull/58377 and the Interactivity Router now announces through the wp-a11y script module since https://github.com/WordPress/gutenberg/pull/65123, so I'm closing. Feel free to reopen if something is still missing.
